api4/emoji.go

```diff
@@ -38,13 +38,13 @@ func createEmoji(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if r.ContentLength > app.MaxEmojiFileSize {
-		c.Err = model.NewAppError("createEmoji", "api.emoji.create.too_large.app_error", nil, "", http.StatusRequestEntityTooLarge)
+	if err := r.ParseMultipartForm(app.MaxEmojiFileSize); err != nil {
```

Comment:
The `app.MaxEmojiFileSize` limits the amount of memory read, so we're protected there, but the implementation appears to use `io.CopyN` under the covers and looks like it could still return `ErrMessageTooLarge` in some cases.

If we know up front that the content length is too big, should we just discard the output instead? And then return the error code as usual? Looks like we have a helper method `consumeAndClose` that effectively does a `io.Copy(ioutil.Discard, resp.Body)` on a `http.Request`.

Reply:
Good point, I like the idea of "consumeAndClose". Anyway the ErrMessageTooLarge only happens if the not-file data of the POST is too big, it would imply a huge emoji name. For files, if they don't fit in the limited memory it would use the filesystem to store it (not memory). I think it can happen, but not with a accidental misuse of the API.

Reply:
Now the solution works better, actually, it close the Body for any fail, not only for the MaxEmojiFileSize error. :)